src/android/FirebaseMessagingPlugin.java

```diff
@@ -44,6 +47,8 @@ public class FirebaseMessagingPlugin extends ReflectiveCordovaPlugin {
     private NotificationManager notificationManager;
     private FirebaseMessaging firebaseMessaging;
 
+    private FirebaseMessagingActivity firebaseMessagingActivity = new FirebaseMessagingActivity();
```

Comment:
Better to keep launcher inline. You need to store private callbackContext in order to call it after getting feedback from user
```java
private final ActivityResultLauncher<String> requestPermissionLauncher =
        registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
            if (isGranted) {
                // FCM SDK (and your app) can post notifications.
                requiestPermissionCallback.success();
            } else {
                requiestPermissionCallback.error("Notifications permission is not granted");
            }
        });
```

Reply:
Yea I was just trying to get it working first, then was going to propogate the callback

Reply:
@chemerisuk So this is why I had that extra activity before, if we do it inline then 
`registerForActivityResult` cannot be resolved. What do I need to import to get it to work?